common/blockchain_util.py

```diff
@@ -91,13 +91,16 @@ def create_transaction_object(self, *positional_inputs, method_name, address, co
         logger.info(f"nonce :: {nonce}")
         logger.info(f"positional_inputs :: {positional_inputs}")
         gas_price = 3 * (self.web3_object.eth.gasPrice)
-        transaction_object = getattr(self.contract_instance.functions, method_name)(
-            *positional_inputs).buildTransaction({
+        options = {
             "from": address,
             "nonce": nonce,
             "gasPrice": gas_price,
             "chainId": net_id
-        })
+        }
+        if bool(gas):
```

Comment:
why casting to bool and checking. better put
```python
gas is not None:
```

Reply:
This is faster `"gas is not None"` over bool() operation.